tests/mock_unit/test_oob_telemetry.py

```diff
@@ -28,15 +29,18 @@ def test_unit_oob_connection_telemetry(caplog, local_testing_telemetry_setup):
     oob_service = LocalTestOOBTelemetryService.get_instance()
     # clean up queue first
     oob_service.export_queue_to_string()
-    with caplog.at_level(logging.DEBUG, logger="snowflake.snowpark.mock._telemetry"):
+    with caplog.at_level(
+        logging.DEBUG, logger="snowflake.snowpark.mock._telemetry"
+    ), mock.patch("snowflake.snowpark.mock._telemetry.datetime") as mock_datetime:
```

Comment:
nit: here and below, can we only put the `mock.patch` around the method calls that require it? that way it's clearer why we're doing the patching, the diff is smaller, and it's easier to understand the indentation below. I think that might mean just putting the context around the `log_session_creation`.

Reply:
sure thing. I updated the ctx mgr at a reasonable granularity and added comments.